pysensors/utils/_validation.py

```diff
@@ -0,0 +1,48 @@
+"""
+Various utility functions for validation and computing reconstruction scores and errors.
+"""
+import numpy as np
+
+def determinant(top_sensors, n_features, basis_matrix):
+    """
+    Function for calculating |C.T phi.T C phi|.
+
+    Parameters
+        ----------
+        top_sensors: np.darray,
+            Column indices of choosen sensor locations
+        n_features : int,
+            No. of features of dataset
+        basis_matrix : np.darray,
+            The basis matrix calculated by model.basis_matrix_
+        
+        Returns
+        -------
+        optimality : Float, 
+            The dterminant value obtained.
+    """
+    
+    c = np.zeros([len(top_sensors),n_features])
+    for i in range(len(top_sensors)):
+        c[i,top_sensors[i]] = 1
+    phi = basis_matrix
+    optimality = np.linalg.det((c@phi).T @ (c@phi)) #np.log(np.linalg.det(phi.T @ c.T)) np.log(np.linalg.det((c@phi).T @ (c@phi)))
```

Comment:
Based on the number of sensors (oversampling) the log determinant formula is different.

Reply:
Same as above